src/JSInterop/Microsoft.JSInterop/test/DotNetDispatcherTest.cs

```diff
@@ -574,6 +575,27 @@ protected override string InvokeJS(string identifier, string argsJson)
                 _nextInvocationTcs = new TaskCompletionSource<object>();
                 return null;
             }
+
+            protected internal override void EndInvokeDotNet(
+                string callId,
+                bool success,
+                object resultOrError,
+                string assemblyName,
+                string methodIdentifier,
+                long dotNetObjectId)
+            {
+                if (!success && resultOrError is Exception ex)
+                {
+                    resultOrError = ex.ToString();
+                }
+                else if (!success && resultOrError is ExceptionDispatchInfo edi)
+                {
+                    resultOrError = edi.SourceException.ToString();
+                }
+
+                var args = JsonSerializer.Serialize(new[] { callId, success, resultOrError }, new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase });
+                BeginInvokeJS(0, "DotNet.jsCallDispatcher.endInvokeDotNetFromJS", args);
```

Comment:
This seems super weird for a test class to be doing. It's pretty misleading to put production-like code in a test fake, as it suggests to test code that this behavior is inherent (e.g., in the base class). It's exceedingly strange for the unit tests to be making assertions that it does this JS interop call when that's not inherent to the contract any more.

Instead of doing this, could we have it behave similarly to the other overrides on the test class, i.e., log the args and let the unit tests make assertions about them?